### PR TITLE
introduce ppc64 big endian glibc target

### PR DESCRIPTION
**Pull request by q66**
This is mostly a work of @pullmoll with some minor fixes contributed by me. It only covers the basics, fixes for LLVM and other packages will come separately.

### CODE REVIEW

File: srcpkgs/gcc/template [hunk 1]
```diff
@@ -116,7 +117,7 @@ case "$XBPS_TARGET_MACHINE" in
 		subpackages+=" libquadmath libquadmath-devel"
 		subpackages+=" libvtv libvtv-devel"
 		;;
-	ppc64le)
+	ppc64|ppc64le)
```
Comment (pullmoll) [inline srcpkgs/gcc/template]: This is actually wrong. It was a try but with `libquadmath` enabled I cannot cross compile gcc from x86_64.
It is not that this is actually required AFAICT so please remove the `ppc64|` again.

File: srcpkgs/gcc/template [hunk 2]
```diff
@@ -381,7 +382,7 @@ do_install() {
 	done
 
 	case "$XBPS_TARGET_MACHINE" in
-		i686*|x86_64*|ppc64le) ;;
+		i686*|x86_64*|ppc64|ppc64le) ;;
```
Comment (pullmoll) [inline srcpkgs/gcc/template]: Same here.

File: srcpkgs/gcc/template (hunk 1 above)
Comment (q66) [reply srcpkgs/gcc/template]: Alright. I'm recompiling gcc right now to investigate the cause of that, but I'll probably end up removing it then.

**Comment by pullmoll** [comment]
The error I get when cross compiling from x86_64 using cross-powerpc64-linux-gnu is:
```
=> libquadmath-8.3.0_1: running pkg_install ...
mv: cannot stat '/destdir/powerpc64-linux-gnu/gcc-8.3.0/usr/lib/libquadmath.so*': No such file or directory
=> ERROR: libquadmath-8.3.0_1: pkg_install: 'mv ${_destdir}/$files ${_pkgdestdir}/${_targetdir}' exited with 1
=> ERROR:   in _vmove() at common/environment/setup/install.sh:232
=> ERROR:   in _noglob_helper() at common/environment/setup/install.sh:12
=> ERROR:   in pkg_install() at srcpkgs/libquadmath/template:534
```

**Comment by q66** [comment]
I guess that actually makes sense, as the quad-precision stuff in glibc requires POWER8 and higher, which we don't compile for, and libquadmath will probably depend on that. Okay noted, after the compilation fails, I will just remove it, and put in `--disable-libquadmath` just for clarity.

**Comment by q66** [comment]
(in musl it's disabled because it does not support 128-bit `long double` on any CPU)

**Comment by q66** [comment]
```
=> Registering new packages to /host/binpkgs/be
index: added `base-voidstrap-0.9_2' (ppc64).
index: 232 packages registered.
```

**Comment by pullmoll** [comment]
Could you try if you can build e.g. `atk` or `gdk-pixbuf`? This kind of pkgs still fail here with this error:
```
/bin/bash: error while loading shared libraries: /usr/powerpc64-linux-gnu/usr/lib/libdl.so.2: ELF file data encoding not little-endian
linking of temporary binary failed: Command '['powerpc64-linux-gnu-gcc', '-o', '/builddir/gdk-pixbuf-2.38.1/build/tmp-introspectz9jqnmyz/GdkPixbuf-2.0', '-fstack-clash-protection', '-D_FORTIFY_SOURCE=2', '-O2', '-pipe', '-mcpu=970', '-mtune=power9', '-maltivec', '-mabi=elfv2', '-g', '-I/usr/powerpc64-linux-gnu/usr/include', '/builddir/gdk-pixbuf-2.38.1/build/tmp-introspectz9jqnmyz/GdkPixbuf-2.0.o', '-L.', '-Wl,-rpath,.', '-Wl,--no-as-needed', '-L/builddir/gdk-pixbuf-2.38.1/build/gdk-pixbuf', '-Wl,-rpath,/builddir/gdk-pixbuf-2.38.1/build/gdk-pixbuf', '-L/usr/powerpc64-linux-gnu/usr/lib', '-Wl,-rpath,/usr/powerpc64-linux-gnu/usr/lib', '-L/usr/powerpc64-linux-gnu/usr/lib', '-Wl,-rpath,/usr/powerpc64-linux-gnu/usr/lib', '-lgdk_pixbuf-2.0', '-lm', '-lgobject-2.0', '-lglib-2.0', '-lgmodule-2.0', '-lgio-2.0', '-lgio-2.0', '-lgobject-2.0', '-Wl,--export-dynamic', '-pthread', '-lgmodule-2.0', '-lglib-2.0', '-Wl,-z,relro', '-Wl,-z,now', '-Wl,--as-needed', '-L/usr/powerpc64-linux-gnu/usr/lib']' returned non-zero exit status 127.
ninja: build stopped: subcommand failed.
=> ERROR: gdk-pixbuf-2.38.1_1: do_build: '${make_cmd} -C ${meson_builddir} ${makejobs} ${make_build_args} ${make_build_target}' exited with 1
=> ERROR:   in do_build() at common/build-style/meson.sh:134
```

**Comment by q66** [comment]
trying right now...

**Comment by q66** [comment]
looks like i'll need to do the llvm patch bits first, it carries that in somewhere on the way

**Comment by pullmoll** [comment]
Oh, then llvm may be the source of this odd message. We can solve this later on, after merging this PR. If noone objects I'll merge it once the builders are idle after the kde5 updates.

**Comment by q66** [comment]
yeah, llvm will need to be patched and also revbumped, as llvm is a cross compiler, so it needs to be treated the same as a crosstoolchain

i don't think llvm is actually the source of the error here, looks like something to do with gobject-introspection, but yes, hardly relevant to the changeset, it can be fixed later

**Pull request closed without merging**